src/praisonai/praisonai/agents_generator.py

```
@@ -456,27 +460,32 @@ def generate_crew_and_kickoff(self):
+                    'ScrapeWebsiteTool', 'WebsiteSearchTool', 'XMLSearchTool',
+                    'YoutubeChannelSearchTool', 'YoutubeVideoSearchTool',
+                ]
+                
+                # Resolve only tools that are actually available
+                for tool_name in standard_tools:
+                    if tool_name in available_tools:
+                        resolved_tool = self.tool_resolver.resolve(tool_name)
+                        if resolved_tool is not None:
+                            tools_dict[tool_name] = resolved_tool() if inspect.isclass(resolved_tool) else resolved_tool
+                            
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)